eip_operator/src/controller/node.rs

```diff
@@ -88,14 +131,17 @@ impl k8s_controller::Context for Context {
             client.clone(),
             self.namespace.as_deref().unwrap_or("default"),
         );
-
-        let node_labels = node.labels().ok_or(Error::MissingNodeLabels)?;
-        let all_eips = eip_api.list(&ListParams::default()).await?.items;
-        let eip = all_eips
-            .into_iter()
-            .filter(|eip| eip.attached())
-            .find(|eip| eip.matches_node(node_labels));
-        if let Some(eip) = eip {
+        let node_labels = node.labels();
+        let matched_eips = eip_api.list(&ListParams::default()).await?.items;
```

Comment:
Nitpick, this is all EIPs, not just matched ones. Perhaps the ownership rules would allow you to just chain the `into_iter` here without storing them first (I'm not sure).